contracts/BC_fusion/StakeHub.sol

```diff
@@ -1179,4 +1208,12 @@ contract StakeHub is System, Initializable, Protectable {
         uint256 bnbAmount = IStakeCredit(_validators[operatorAddress].creditContract).claim(msg.sender, requestNumber);
         emit Claimed(operatorAddress, msg.sender, bnbAmount);
     }
+
+    function _getMsgSender() internal view returns (address) {
```

Comment:
It may be better use `_getOperatorFromMsgSender`

Reply:
`_getOperatorFromMsgSender ` may be inaccurate. In most cases, `msg.sender` is returned directly, and in rare cases, the `agent` can appear as a `msg.sender` replacement for the `operatorAddress`.

Reply:
`_getOperatorFromMsgSender ` may be inaccurate. In most cases, `msg.sender` is returned directly, and in rare cases, the `agent` can appear as a `msg.sender` replacement for the `operatorAddress`.

Reply:
_getOperatorByMsgSender？

Reply:
This method is not to get `operator` address,  it's to get the real `msg.sender` for all address. Only in some scenarios, the agent can act as the operator's `msg.sender`.

Reply:
we can refer to the similar code here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/05f218fb6617932e56bf5388c3b389c3028a7b73/contracts/metatx/ERC2771Context.sol#L55